lib/matplotlib/animation.py

```diff
@@ -1196,10 +1196,24 @@ class to use, such as 'ffmpeg'. If ``None``, defaults to
                 for anim in all_anim:
                     # Clear the initial frame
                     anim._init_draw()
-                for data in zip(*[a.new_saved_frame_seq() for a in all_anim]):
+                _truncated = False
+                for i, data in enumerate(
+                        zip(*[a.new_saved_frame_seq() for a in all_anim])):
                     for anim, d in zip(all_anim, data):
+                        if d is _truncated_sentinel:
```

Comment:
Just for my learning, why do you need `_trundated_sentinel` in here, versus just counting to 100?

Reply:
To only trigger the warning in the case the user passed in a generator.  If the user passed in just a very long but explicit list, the previous code would (correctly) not truncate it, and we don't want to trigger a warning in this case.

Reply:
Can't you do this in a clearer way (i.e. w/o `itertools.chain`)?  I guess I'd be OK w/ this construct if it saves a bunch of code, or maybe this is really the only way to do it, but it seems this check could be less opaque.

Reply:
AFAICT I need to tag a sentinel at the end of the generator.  But this needs to be done without actually running the generator first (because that can require arbitrarily large resources); so itertools.chain is literally the way to do that.
Alternatively I would need to keep another flag around saying "this is a generator-based animation, please check whether we reached the 100th frame", which I guess works too but I am not sure it's simpler...

Reply:
I think this is a reasonable way to implement this, it keeps the state all in one place.

Reply:
If this is the best way to do it, it would help if there was a bit of a comment in the code explaining what is going on, in case you both decide to take up extreme crocheting and don't stop around here anymore.

Reply:
Actually I think I figured out a less obfuscated way to implement this...